file-manager/fileManager.cpp

```diff
@@ -0,0 +1,168 @@
+    doc << FILE_PATH << filePath;
+    doc << FILE_DATA << data;
+
+    // Insert the document into the collection.
+    collection.insert_one(doc.view());
+    file.close();
+
+    std::cout << "Upload successful for: " << filePath << std::endl;
+
+    return true;
+}
+
+// Download a file from a collection to a given folder.
+bool download(const std::string& fileName, const std::string& destiationFolder, mongocxx::collection& collection)
+{
+    // Create a query to find the file by filename
+    bsoncxx::builder::stream::document filter;
+    filter << FILE_NAME << fileName;
+    auto filterDoc = filter << bsoncxx::builder::stream::finalize;
+
+     // Find the document in the collection
+    auto result = collection.find_one(filterDoc.view());
+
+    if (result) 
+    {
+        // Get the binary data from the document
+        bsoncxx::document::view binaryDocView = result->view();
+        auto binaryData = binaryDocView[FILE_DATA].get_binary();
+        
+        // Create a file to save the binary data
+        std::ofstream file(destiationFolder + fileName, std::ios::binary);
+        if (!file) 
+        {
+            std::cout << "Failed to create the file: " << fileName << " at " << destiationFolder << std::endl;
+            return false;
+        }
+        
+        // Write the binary data to the file
+        file.write(reinterpret_cast<const char*>(binaryData.bytes), binaryData.size);
+        file.close();
+
+        std::cout << "Download sucessful for: " << fileName << " at " << destiationFolder << std::endl;
+        return true;
+    } 
+    else 
+    {
+        std::cout << "File not found in the collection: " << fileName << std::endl;
+        return false;
+    }
+
+}
+
+
+// ********************************************** Main **********************************************
+int main()
+{
+    try
+    {
+        auto mongoURIStr = "<Insert MongoDB Connection String>";
+        static const mongocxx::uri mongoURI = mongocxx::uri{ mongoURIStr };
+
+        if(mongoURI.to_string().empty())
+        {
+            std::cout << "URI is empty";
+            return 0;
+        }
+        
+        // Create an instance.
+        mongocxx::instance inst{};
+        
+        mongocxx::options::client client_options;
+        auto api = mongocxx::options::server_api{ mongocxx::options::server_api::version::k_version_1 };
+        client_options.server_api_opts(api);
+        mongocxx::client conn{ mongoURI, client_options};
+        
+        const std::string dbName = "fileStorage";
+        const std::string collName = "files";
+        
+        auto fileStorageDB = conn.database(dbName);
+        auto filesCollection = fileStorageDB.collection(collName);
+
+        // Upload all files in the upload folder.
+        const std::string uploadFolder = "/Users/bishtr/repos/fileStorage/upload/";
+        for (const auto & filePath : std::filesystem::recursive_directory_iterator(uploadFolder))
```

Comment:
As implemented, downloading flattens the directory tree of the files uploaded.

Example:

Upload this directory:
```
upload_from
├── a.txt
├── b.txt
└── dir
    └── a.txt
```

Downloading back results in:
```
download_to
├── a.txt
└── b.txt
```

Only one of the "a.txt" entries is in the results. And the `dir` directory is lost.

Is this OK for this example?

If recursively uploading is not central to this example, I suggest replacing the `recursive_directory_iterator` with a `directory_iterator`.
```suggestion
        for (const auto & filePath : std::filesystem::directory_iterator(uploadFolder))
```